src/main/java/net/minecraftforge/network/HandshakeHandler.java

```diff
@@ -329,6 +331,78 @@ void handleConfigSync(final HandshakeMessages.S2CConfigData msg, final Supplier<
         contextSupplier.get().setPacketHandled(true);
         NetworkConstants.handshakeChannel.reply(new HandshakeMessages.C2SAcknowledge(), contextSupplier.get());
     }
+
+    void handleClientReset(HandshakeMessages.S2CReset msg, final Supplier<NetworkEvent.Context> contextSupplier) {
+        NetworkEvent.Context context = contextSupplier.get();
+        Connection connection = context.getNetworkManager();
+
+        if (context.getDirection() != NetworkDirection.LOGIN_TO_CLIENT && context.getDirection() != NetworkDirection.PLAY_TO_CLIENT) {
+            connection.disconnect(Component.literal("Illegal packet received, terminating connection"));
+            throw new IllegalStateException("Invalid packet received, aborting connection");
+        }
+
+        LOGGER.debug(FMLHSMARKER, "Received reset from server");
+
+        ServerData serverData = Minecraft.getInstance().getCurrentServer();
+        Screen screen = Minecraft.getInstance().screen;
+
+        if (!handleClear(context)) {
+            return;
+        }
+
+        NetworkHooks.registerClientLoginChannel(connection);
+        connection.setProtocol(ConnectionProtocol.LOGIN);
+        connection.setListener(new ClientHandshakePacketListenerImpl(
+                connection, Minecraft.getInstance(), serverData, screen, true, null, statusMessage -> {}
+        ));
+
+        context.setPacketHandled(true);
+        NetworkConstants.handshakeChannel.reply(
+                new HandshakeMessages.C2SAcknowledge(),
+                new NetworkEvent.Context(connection, NetworkDirection.LOGIN_TO_CLIENT, 98)
+        );
+
+        LOGGER.debug(FMLHSMARKER, "Reset complete");
+    }
+
+    @OnlyIn(Dist.CLIENT)
```

Comment:
Usage of OnlyIn to strip new code is against guidelines.

Client-only code should be in a client-only class, the annotation is only for use marking vanilla code as client-only.

Reply:
I wasn't sure if there was a class that I should move them to but searching through the package they were in I couldn't find any other classes that were explicitly client only, so I moved them to the ClientResetPacketHandler class for the time being